db/db_impl/db_impl.cc

```diff
@@ -685,12 +694,49 @@ void DBImpl::PersistStats() {
     stats_history_size_limit = mutable_db_options_.stats_history_buffer_size;
   }
 
-  // TODO(Zhongyi): also persist immutable_db_options_.statistics
-  {
-    std::map<std::string, uint64_t> stats_map;
-    if (!statistics->getTickerMap(&stats_map)) {
-      return;
+  std::map<std::string, uint64_t> stats_map;
+  if (!statistics->getTickerMap(&stats_map)) {
+    return;
+  }
+
+  if (immutable_db_options_.persist_stats_to_disk) {
+    WriteOptions wo;
+    wo.low_pri = true;
+    wo.no_slowdown = true;
+    wo.sync = false;
+    wo.disableWAL = true;
+    WriteBatch batch;
+    if (stats_slice_initialized_) {
+      // write version key
+      char version_key[100];
+      int version_key_length = PersistentStatsHistoryIterator::EncodeVersionKey(
+          now_micros, 100, version_key);
```

Comment:
You are doing a version key per stats. Do you consider to have one single version key for the whole CF?